src/window_handle.rs

```diff
@@ -748,11 +774,40 @@ impl WindowHandle {
                         if cx.app_state.focus != Some(id) {
                             let old = cx.app_state.focus;
                             cx.app_state.focus = Some(id);
+
+                            // Fix: Focus events were never dispatched to views' own
+                            // event_before_children / event_after_children.
+                            //
+                            // This is a bit messy; could it be done in ViewId.apply_event()
+                            // instead?  Are there other cases of dropped events?
```

Comment:
Other events are from winit, so they would be dispatched by the `event` method, while `Event::FocusGained` and `Event::FocusLost` is kind of an "internal" event.